File: backend/ai/views/chat_conversation.py
from rest_framework import serializers

from ai.models import ChatConversation
from utils.serializers import CustomModelSerializer
from utils.custom_model_viewSet import CustomModelViewSet
from django_filters import rest_framework as filters


class ChatConversationSerializer(CustomModelSerializer):
    username = serializers.CharField(source='user.username', read_only=True)

    """
    AI 聊天对话 序列化器
    """
    class Meta:
        model = ChatConversation
        fields = '__all__'
        read_only_fields = ['id', 'create_time', 'update_time']

class ChatConversationFilter(filters.FilterSet):

    class Meta:
        model = ChatConversation
        fields = ['id', 'remark', 'creator', 'modifier', 'is_deleted', 'title', 'pinned', 'model',
                  'system_message', 'max_tokens', 'max_contexts']


class ChatConversationViewSet(CustomModelViewSet):
    """
    AI 聊天对话 视图集
    """
    queryset = ChatConversation.objects.filter(is_deleted=False).order_by('-id')
    serializer_class = ChatConversationSerializer
    filterset_class = ChatConversationFilter
    search_fields = ['name']  # 根据实际字段调整
    ordering_fields = ['create_time', 'id']
    ordering = ['-create_time']

# 移入urls中

File: web/apps/web-antd/src/models/ai/chat_conversation.ts
import { BaseModel } from '#/models/base';

export namespace AiChatConversationApi {
  export interface AiChatConversation {
    id: number;
    remark: string;
    creator: string;
    modifier: string;
    update_time: string;
    create_time: string;
    is_deleted: boolean;
    title: string;
    pinned: boolean;
    pinned_time: string;
    user: number;
    role: number;
    model_id: number;
    model: string;
    system_message: string;
    temperature: any;
    max_tokens: number;
    max_contexts: number;
  }
}

export class AiChatConversationModel extends BaseModel<AiChatConversationApi.AiChatConversation> {
  constructor() {
    super('/ai/chat_conversation/');
  }
}
